File: __tests__/MiniGetterSetter.py
class MiniGetterSetter:
    def __init__(self):
        self.vals = {}
    
    def set_value(self, args,settings):
        assert 'key' in args
        assert 'val' in args
        print("setting value "+args['val'])
        self.vals[args['key']] = args['val']
        return True


    def get_value(self, args,settings):
        if args['key'] in self.vals:
            print("getting value "+args['key'])
            return self.vals[args['key']]
        return self.vals
    
    def get_all(self, args,settings):
        return self.vals    

File: __tests__/nodejs/MiniGetterSetter.js
class MiniGetterSetter {
    constructor() {
        this.vals = {};
    }

    set_value(args, settings) {
        if (!args.hasOwnProperty('key') || !args.hasOwnProperty('val')) {
            throw new Error('Missing required arguments: key and/or val');
        }
        console.log('setting value ' + args.val);
        this.vals[args.key] = args.val;
        return true;
    }

    get_value(args, settings) {
        if (this.vals.hasOwnProperty(args.key)) {
            console.log('getting value ' + args.key);
            return this.vals[args.key];
        }
        return this.vals;
    }

    get_all(args, settings) {
        return this.vals;
    }
}

export default  MiniGetterSetter;
export {MiniGetterSetter};
